src/main/java/com/bp3x/raidbot/commands/lfg/LFGCommand.java

```diff
@@ -1,135 +1,139 @@
+        String shortName = commandEvent.optString("event");
+        String date = commandEvent.optString("date");
+        String time = commandEvent.optString("time");
+    
+        var commandUser = commandEvent.getMember().getUser();
+        log.info("LFG command by: " + commandUser.getName() + "#" + commandUser.getDiscriminator());
+        log.info("Wants to schedule " + shortName + " " + date + " " + time);
+        
+        commandEvent.deferReply(true).queue(hook -> {
+            // get and verify the user's timezone role for conversion
+            ArrayList<Role> timezoneRoles = RaidBot.getConfig().getTimezoneRoles();
+            Role userTimezoneRole = commandEvent.getMember().getRoles().stream().filter(timezoneRoles::contains).findFirst().orElse(null);
+            if (userTimezoneRole == null) {
+                hook.editOriginal("You do not have a timezone role assigned. Ask an admin to give you your role.").queue();
                 return;
             }
-        } catch (DateTimeParseException e) {
-            MessageUtils.sendAutoDeletedMessage("Invalid date/time argument.", 300, commandEvent);
-            return;
-        }
-
-        try {
-            if (Event.eventExists(activityString)) {
-                Event plannedEvent = new Event(activityString, eventDateTime);
-                plannedEvent.setPlayerStatus(commandEvent.getMember(), Event.EventPlayerStatus.ACCEPTED);
-                LFGEmbedBuilder builder = new LFGEmbedBuilder(plannedEvent);
-                MessageCreateBuilder messageCreateBuilder = new MessageCreateBuilder();
-                messageCreateBuilder.addEmbeds(builder.build());
-
-                Message success = commandEvent.getChannel().sendMessage(messageCreateBuilder.build()).complete();
-
-                RestAction<Void> reactWhiteCheckMark = success.addReaction(LFGConstants.ACCEPTED_EMOJI);
-                RestAction<Void> reactQuestion = success.addReaction(LFGConstants.TENTATIVE_EMOJI);
-                RestAction<Void> reactCross = success.addReaction(LFGConstants.DECLINED_EMOJI);
-
-                RestAction.allOf(reactWhiteCheckMark, reactQuestion, reactCross).queue();
-
-                plannedEvent.registerEvent(success);
-
-                Event.scheduleEventDeletion(eventDateTime, commandEvent, success);
-
-                MessageUtils.autoDeleteMessage(commandEvent.getMessage(), 300);
-
-                success.createThreadChannel(plannedEvent.getLongName() + "-" + plannedEvent.getEventId()).queue(thread ->
-                {
-                    thread.sendMessage("Creating this thread to help you organize your event").queue();
-                    thread.addThreadMember(commandEvent.getAuthor()).queue();
-                });
-
-            } else {
-                MessageUtils.sendAutoDeletedMessage("That event does not exist", 300, commandEvent);
+            
+            if (commandEvent.getChannelType().isThread()) {
```

Comment:
Nice check, don't think I would have thought of this.

Reply:
Only caught this because I accidentally found myself trying to execute commands within an event thread while testing 😂